core-web/libs/block-editor/src/lib/extensions/indent/indent.extension.ts

```diff
@@ -0,0 +1,225 @@
+    less = -40
+}
+
+function setNodeIndentMarkup(tr: Transaction, pos: number, delta: number): Transaction {
+    if (!tr.doc) return tr;
+
+    const node = tr.doc.nodeAt(pos);
+    if (!node) return tr;
+
+    const minIndent = IndentProps.min;
+    const maxIndent = IndentProps.max;
+    const indent = clamp((node.attrs.indent || 0) + delta, minIndent, maxIndent);
+
+    if (indent === node.attrs.indent) return tr;
+
+    const nodeAttrs = {
+        ...node.attrs,
+        indent
+    };
+
+    return tr.setNodeMarkup(pos, node.type, nodeAttrs, node.marks);
+}
+
+function updateIndentLevel(tr: Transaction, delta: number): Transaction {
+    const { doc, selection } = tr;
+
+    if (!doc || !selection) return tr;
+
+    if (!(selection instanceof TextSelection || selection instanceof AllSelection)) {
+        return tr;
+    }
+
+    const { from, to } = selection;
+
+    doc.nodesBetween(from, to, (node, pos) => {
+        const nodeType = node.type;
+
+        // Only handle paragraphs, headings and blockquotes - NO lists
+        if (
+            nodeType.name === 'paragraph' ||
+            nodeType.name === 'heading' ||
+            nodeType.name === 'blockquote'
+        ) {
+            tr = setNodeIndentMarkup(tr, pos, delta);
+            return false;
+        }
+
+        return true;
+    });
+
+    return tr;
+}
+
+export const IndentExtension = Extension.create<IndentOptions>({
+    name: 'indent',
+
+    addOptions() {
+        return {
+            types: ['heading', 'paragraph', 'blockquote'],
+            minIndentLevel: 0,
+            maxIndentLevel: 10,
+            indentSize: 40
+        };
+    },
+
+    addGlobalAttributes() {
+        return [
+            {
+                types: this.options.types,
+                attributes: {
+                    indent: {
+                        default: 0,
+                        renderHTML: (attributes) => {
+                            if (!attributes.indent || attributes.indent <= 0) {
+                                return {};
+                            }
+                            return {
+                                style: `margin-left: ${attributes.indent}px;`
+                            };
+                        },
+                        parseHTML: (element) => {
+                            const marginLeft = element.style.marginLeft;
+                            if (!marginLeft) return 0;
+
+                            const value = parseInt(marginLeft, 10);
+                            return value
+                                ? Math.round(value / this.options.indentSize) *
+                                      this.options.indentSize
+                                : 0;
+                        }
+                    }
+                }
+            }
+        ];
+    },
+
+    addCommands() {
+        return {
+            indent:
+                () =>
+                ({ tr, state, dispatch, editor }) => {
```

Comment:
'tr' and editor are declared, but their value is never read; please just use state and dispatch in this case.